packages/cdc-react/src/components/ProfileHeader/ProfileHeaderUserProfileMenuItem/ProfileHeaderUserProfileMenuItem.tsx

```diff
@@ -0,0 +1,166 @@
+    },
+    Enter: (currentButton?: HTMLButtonElement) => {
+      currentButton?.click();
+    },
+  } as const;
+
+  const handlePopupKeyDown = (ev: React.KeyboardEvent) => {
+    ev.preventDefault();
+
+    const popupMenuButtons: HTMLCollectionOf<Element> | HTMLButtonElement[] =
+      document.getElementsByClassName("profile-header-popup-menu-item");
+
+    const currentButton: HTMLButtonElement = popupMenuButtons[
+      activePopupItemIndex
+    ] as HTMLButtonElement;
+
+    const isLastIndex: boolean =
+      activePopupItemIndex === popupMenuButtons.length - 1;
+
+    const code: string = ev.code;
+    const keys: string[] = Object.keys(PopupKeyDownMethods);
+
+    if (popupOpen && keys.includes(code)) {
+      PopupKeyDownMethods[code](currentButton, isLastIndex);
+    }
+  };
+
+  useLayoutEffect(() => {
+    const popupMenuButtons: HTMLCollectionOf<Element> | HTMLButtonElement[] =
+      document.getElementsByClassName("profile-header-popup-menu-item");
+
+    const buttonToFocus: HTMLButtonElement = popupMenuButtons[
+      activePopupItemIndex
+    ] as HTMLButtonElement;
+
+    buttonToFocus.focus();
+  });
+
+  useEffect(() => {
+    document.addEventListener(
+      "mousedown",
+      handleClickOutsidePopup as unknown as EventListenerOrEventListenerObject,
+      true
+    );
+
+    return () => {
+      document.removeEventListener(
+        "mousedown",
+        handleClickOutsidePopup as unknown as EventListenerOrEventListenerObject,
+        true
+      );
+    };
+  });
+
+  return (
+    <>
+      <span className="sr-only">User profile button</span>
+
+      <button className={className} ref={userProfileMenuItemRef} tabIndex={0}>
+        <svg
+          xmlns="http://www.w3.org/2000/svg"
+          width="36"
+          height="36"
+          viewBox="0 0 40 40"
+          fill="#DDD"
+          className="user-image"
+          stroke="#ddd">
+          <circle cx="20" cy="20" r="20"></circle>
+        </svg>
+        <Icon
+          name={popupOpen ? "chevron-up" : "chevron-down"}
+          className={popupOpen ? "chevron-up" : "chevron-down"}
+        />
+      </button>
+
+      <div
+        className={`profile-header-popup-wrap ${
+          popupOpen ? "profile-header-popup-wrap-show" : ""
+        }`}
+        onKeyDown={handlePopupKeyDown}>
+        <div className="profile-header-popup" ref={popupRef}>
+          {(userProfilePopupMenuItems || []).map((popupMenuItem, index) => (
+            <ProfileHeaderPopupMenuItem
+              key={`profile-header-popup-menu-item-${index}`}
+              className={`profile-header-popup-menu-item ${popupMenuItem.className} profile-header-popup-menu-item${index}`}
+              icon={popupMenuItem.icon}
+              iconPosition={popupMenuItem.iconPosition}
+              text={popupMenuItem.text}
+              badgeCount={popupMenuItem.badgeCount}
+              onClick={popupMenuItem.onClick}
+            />
+          ))}
+        </div>
+      </div>
```

Comment:
Should this be conditionally rendered based on the state of `popupOpen`?

Reply:
I think the tertiary condition on line 147 is taking care of this.  But might be able to rely on the state of `popupOpen` instead of the class and get rid of the class.